File: leetcode/rust/largest-local-values-in-a-matrix.rs
impl Solution {
    pub fn largest_local(grid: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let n = grid.len();
        let mut result = Vec::new();

        for i in 0..=n - 3 {
            let mut row = Vec::new();

            for j in 0..=n - 3 {
                let mut max = 0;

                for x in 0..3 {
                    for y in 0..3 {
                        max = max.max(grid[i + x][j + y]);
                    }
                }

                row.push(max);
            }

            result.push(row);
        }

        result
    }
}

File: leetcode/python/largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: list[list[int]]) -> list[list[int]]:
        n = len(grid)
        result = []

        for i in range(n - 2):
            row = []

            for j in range(n - 2):
                max_n = 0

                for x in range(3):
                    for y in range(3):
                        max_n = max(max_n, grid[i + x][j + y])
                
                row.append(max_n)

            result.append(row)

        return result
        